### DISCUSSION TITLE
I want container queries

### DISCUSSION THREAD

**Discussion by aarongustafson** [WebWeWant/webwewant.fyi#118] [Wants]
Currently, elements can only be sized based on the size of the viewport. However, there&#39;s often the need to size an element based on the size of its parent container irrespective of the viewport size. The [use cases are really well documented](https://wicg.github.io/cq-usecases/) and [Resize Observer](https://developer.mozilla.org/docs/Web/API/ResizeObserver) isn&#39;t sufficient to address them.

One challenge in this area is that implementing container queries could cause infinite loops if the size of child elements alter the size of the container which then changes the size of the child elements and so on.

<hr>

https://webwewant.fyi/wants/22/

**Comment by ststimac** [comment]
Intent to prototype container queries from the Chrome team: https://groups.google.com/a/chromium.org/g/blink-dev/c/u1AKdrXhPGI/m/wrJb-unhAgAJ?pli=1

**Reply by aarongustafson** [reply]
Interesting approach. Another is from @bkardell: https://bkardell.com/blog/TowardResponsive.html

**Comment by aarongustafson** [comment]
Alerting @davatron5000, @b-strauss, @mzampetti, and @segunolalive who all requested this.

**Reply by segunolalive** [reply]
Thanks @aarongustafson
